scrapy/utils/reqser.py

```diff
@@ -58,7 +59,8 @@ def request_from_dict(d, spider=None):
         meta=d['meta'],
         encoding=d['_encoding'],
         priority=d['priority'],
-        dont_filter=d['dont_filter'])
+        dont_filter=d['dont_filter'],
+        flags= d['flags'])
```

Comment:
I think it should be d.get('flags'), for backwards compatibility, i.e. to be able to continue a long crawl after Scrapy update. 

Maybe it makes sense not to serialize flags if flags is None, but I'm not sure about that.